### PR TITLE
Allow setting customizing the outgoing messages by setting AMQP attributes just before dispatching the message to the broker

### PR DESCRIPTION
**Pull request by SzymonPobiega**
Port of https://github.com/Particular/NServiceBus.RabbitMQ/pull/1468 to release 9.2

### CODE REVIEW

**Review by WilliamBZA — APPROVED** [review]

File: src/NServiceBus.Transport.RabbitMQ/RabbitMQTransport.cs [hunk 1]
```diff
@@ -91,6 +92,21 @@ public TimeSpan TimeToWaitBeforeTriggeringCircuitBreaker
             }
         }
 
+        /// <summary>
+        /// Gets or sets the action that allows customization of the native <see cref="IBasicProperties"/> 
+        /// just before it is dispatched to the rabbitmq client. 
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This customization is applied after any configured transport customizations, meaning that
```
Comment (tmasternak) [inline src/NServiceBus.Transport.RabbitMQ/RabbitMQTransport.cs:101]:
```suggestion
        /// When provided the customization is called after all other transport customizations configured, meaning that 
```

File: src/NServiceBus.Transport.RabbitMQ/RabbitMQTransport.cs [hunk 2]
```diff
@@ -91,6 +92,21 @@ public TimeSpan TimeToWaitBeforeTriggeringCircuitBreaker
...
         }
 
+        /// <summary>
+        /// Gets or sets the action that allows customization of the native <see cref="IBasicProperties"/> 
+        /// just before it is dispatched to the rabbitmq client. 
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This customization is applied after any configured transport customizations, meaning that 
+        /// any changes made here may override or conflict with previous transport-level adjustments.
```
Comment (tmasternak) [inline src/NServiceBus.Transport.RabbitMQ/RabbitMQTransport.cs:102]:
```suggestion
        /// any changes made by customization logic may override or conflict with previous transport-level adjustments. 
```

File: src/NServiceBus.Transport.RabbitMQ/RabbitMQTransport.cs [hunk 3]
```diff
@@ -91,6 +92,21 @@ public TimeSpan TimeToWaitBeforeTriggeringCircuitBreaker
...
 
+        /// <summary>
+        /// Gets or sets the action that allows customization of the native <see cref="IBasicProperties"/> 
+        /// just before it is dispatched to the rabbitmq client. 
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This customization is applied after any configured transport customizations, meaning that 
+        /// any changes made here may override or conflict with previous transport-level adjustments. 
+        /// Exercise caution, as modifying the message at this stage can lead to unintended behavior
```
Comment (tmasternak) [inline src/NServiceBus.Transport.RabbitMQ/RabbitMQTransport.cs:103]:
```suggestion
        /// This extension point should be used with caution, as modifying a native message at this stage can lead to unintended behavior 
```

File: src/NServiceBus.Transport.RabbitMQ/RabbitMQTransport.cs [hunk 4]
```diff
@@ -91,6 +92,21 @@ public TimeSpan TimeToWaitBeforeTriggeringCircuitBreaker
...
+        /// <summary>
+        /// Gets or sets the action that allows customization of the native <see cref="IBasicProperties"/> 
+        /// just before it is dispatched to the rabbitmq client. 
+        /// </summary>
+        /// <remarks>
+        /// <para>
+        /// This customization is applied after any configured transport customizations, meaning that 
+        /// any changes made here may override or conflict with previous transport-level adjustments. 
+        /// Exercise caution, as modifying the message at this stage can lead to unintended behavior 
+        /// downstream if the message structure or properties are altered in ways that do not align
```
Comment (tmasternak) [inline src/NServiceBus.Transport.RabbitMQ/RabbitMQTransport.cs:104]:
```suggestion
        /// downstream if the message content or properties are altered in ways that do not align 
```

**Review by tmasternak — APPROVED** [review]

File: src/NServiceBus.Transport.RabbitMQ.AcceptanceTests/When_customizing_outgoing_messages.cs [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+    using AcceptanceTesting;
+    using global::RabbitMQ.Client.Events;
+    using NServiceBus.AcceptanceTests;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NUnit.Framework;
+
+    class When_customizing_outgoing_messages : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_set_value()
```
Comment (WilliamBZA) [inline src/NServiceBus.Transport.RabbitMQ.AcceptanceTests/When_customizing_outgoing_messages.cs:13]: Perhaps this test can be more clearly names?

```suggestion
        public async Task Should_set_native_property_value()
```

**Pull request merged**